keras/layers/recurrent.py

```diff
@@ -58,12 +61,14 @@ def get_output(self, train):
         # Iterate over the first dimension of the x array (=time).
         outputs, updates = theano.scan(
             self._step, # this will be called with arguments (sequences[i], outputs[i-1], non_sequences[i])
-            sequences=x, # tensors to iterate over, inputs to _step
+            sequences=[x, dict(input=padded_mask, taps=[-1])], # tensors to iterate over, inputs to _step
```

Comment:
Incidentally, this `taps=[-1]` is technically unnecessary since a single negative tap will iterate over the input in the same exact way as just passing in the input without taps (see http://deeplearning.net/software/theano/library/scan.html and search for "abuse of notation"). But it maybe makes things more clear to explicitly name the taps anyway?